React/Views/RCTWrapperViewController.m

```diff
@@ -79,6 +79,17 @@ - (void)viewWillAppear:(BOOL)animated
       [bar setTitleTextAttributes:@{NSForegroundColorAttributeName : _navItem.titleTextColor}];
     }
 
+    if (_navItem.shadowHidden) {
+      // Remove the hairline shadow, but maintain translucency in the bar
+      for (UIView *view in bar.subviews) {
```

Comment:
use a recursive approach for robustness: http://jslim.net/blog/2014/02/28/remove-the-1px-shadow-from-uinavigationbar/